### ISSUE TITLE
Test failure in test-unions.cpp

### ISSUE THREAD

**Issue by strager** [hdoc/hdoc#26]
I built hdoc from source and ran its test suite. I got the following test failure:
```
[doctest] doctest version is "2.4.8"
[doctest] run with "--help" for options
===============================================================================
../tests/index-tests/test-unions.cpp:128:
TEST CASE:  Anonymous struct in a union

../tests/index-tests/test-unions.cpp:163: ERROR: CHECK( s.vars[1].type.name == "float[3]" ) is NOT correct!
  values: CHECK( float [3] == float[3] )

===============================================================================
[doctest] test cases:   88 |   87 passed | 1 failed | 0 skipped
[doctest] assertions: 2821 | 2820 passed | 1 failed |
[doctest] Status: FAILURE!
```
I'm using libclang 13 from Nix.

**Comment by strager** [comment]
I tried with libclang 14 and the test passed.

If a specific libclang is required, I think this version requirement should be listed in the documentation. The docs currently say just "hdoc depends on LLVM/Clang and OpenSSL".

**Comment by hdoc** [comment]
Apologies, there was a change in LLVM/Clang between version 13 and version 14 that caused that specific test case to break. We fixed the test case in hdoc 1.3.0 when we upgraded to LLVM 14, though we didn't make mention of it like you found out the hard way. We'll updated the documentation to note which specific version of LLVM hdoc "lives on" at any given time.
